In [3]:
import pandas as pd
import geopandas as gpd
import numpy as np

# Residential
## Residential Buildings

In [6]:
df_resbuild = gpd.read_file('https://opendata.arcgis.com/datasets/8b57e88e62cf4d9791cf5568675d05da_0.geojson')

In [8]:
df_resbuild.columns

Index(['FID', 'BLDG_ID', 'BLDG_Name', 'Str_Number', 'StreetName', 'Address',
       'Area_Ha', 'Area_Acre', 'GFA_RES_m2', 'GFARes_ft2', 'GFA_RET_m2',
       'GFARet_ft2', 'GFA_Tot_m2', 'GFATot_ft2', 'BLDG_Type', 'Tenure_RES',
       'Tenure_RET', 'Storeys', 'RES_Units', 'RET_Units', 'Inst_Beds',
       'Units_Ha', 'Units_Acre', 'FSI', 'Year_Built', 'WardNumber',
       'geometry'],
      dtype='object')

In [12]:
df_resbuild.head().iloc[1]

FID                                                       2
BLDG_ID                                                1856
BLDG_Name                                                  
Str_Number                                             2921
StreetName                                       Derry Rd E
Address                                     2921 Derry Rd E
Area_Ha                                                0.15
Area_Acre                                          0.370658
GFA_RES_m2                                                0
GFARes_ft2                                                0
GFA_RET_m2                                                0
GFARet_ft2                                                0
GFA_Tot_m2                                                0
GFATot_ft2                                                0
BLDG_Type                                     Mixed RES/RET
Tenure_RES                                                 
Tenure_RET                              

In [9]:
df_resbuild.BLDG_Type.unique()

array(['Mixed RES/RET', 'Apartment', 'Residential Institution', 'Plex'],
      dtype=object)

2921 is said to be a mixed residential retail zone, but looking on Google Maps we see the a law office and lounge. Thusly, it cannot be assumed that the type of building always maps exactly to their use, at least when talking about Mixed Use. 

<img src="./RefImages/2921DerryRoadEast.png" width="200"/>


## Townhouses

In [18]:
df_townhouses = gpd.read_file('https://opendata.arcgis.com/datasets/34a39af0d72143a7a74e675fe5fdcbbf_0.geojson')
df_townhouses.columns

Index(['FID', 'COMPLEX_ID', 'CHAR_AREA', 'Z_AREA', 'Str_Number', 'StreetName',
       'Address', 'OP_DES', 'Zoning', 'Cplex_Name', 'Area_Ha', 'Area_Acre',
       'Parking', 'GFA_RES_m2', 'GFARes_ft2', 'GFA_RET_m2', 'GFARet_ft2',
       'GFA_Tot_m2', 'GFATot_ft2', 'Cplex_Type', 'Tenure_RES', 'Storeys',
       'TH_Blocks', 'RES_Units', 'RET_Units', 'Unit_Ha', 'Unit_Acre', 'FSI',
       'Year_Built', 'WardNumber', 'Shape__Area', 'Shape__Length', 'geometry'],
      dtype='object')

In [20]:
df_townhouses.Cplex_Type.unique()

array(['Townhouse', 'Detached', 'Other (Mobile Homes)', 'Semi-Detached'],
      dtype=object)

In [21]:
df_townhouses.OP_DES.unique()

array(['Medium Density', 'Low Density II', 'High Density',
       'Institutional', 'Mixed Use', 'Low  Density I', 'Low Density I',
       'Low Density II, Medium Density', 'Medium Density, HighDensity',
       'Mixed Use, Medium Density'], dtype=object)

There are detached homes, as well as high and low density homes. Let's take a look at those homes in particular. 

In [23]:
df_townhouses[df_townhouses['Cplex_Type'] == 'Detached'].sample()

FID  COMPLEX_ID              CHAR_AREA Z_AREA      Str_Number  \
422  423        1395  Cooksville NHD (West)     15  2165,2175/2130   

                  StreetName                                 Address  \
422  Stavebank Rd/Dickson Rd  2165,2175/2130 Stavebank Rd/Dickson Rd   

             OP_DES Zoning  Cplex_Name  ...  RES_Units  RET_Units  Unit_Ha  \
422  Low  Density I  R1-19  The Colony  ...         40          0       10   

     Unit_Acre       FSI  Year_Built  WardNumber   Shape__Area  Shape__Length  \
422          4  0.223423        2001           7  76315.574219    1351.730864   

                                              geometry  
422  POLYGON ((-79.61740 43.56039, -79.61716 43.560...  

[1 rows x 33 columns]

2165 Stavebank Rd is detached and has a density level of Low Density 1. Looking on Google Maps we see a large low density detached home. 

## Residential Centres

In [4]:
df_res_centres = gpd.read_file("https://opendata.arcgis.com/datasets/04fcf2b85ba04fe5bd706939f0d811d8_0.geojson")

In [5]:
df_res_centres.columns

Index(['FID', 'CENTRE_ID', 'CHAR_AREA', 'OP_DES', 'ZONING', 'Z_AREA',
       'N_OF_BLDGS', 'PARKING', 'Shape__Area', 'Shape__Length', 'geometry'],
      dtype='object')

In [10]:
df_res_centres['OP_DES'].unique()

array(['High Density', 'Medium Density', 'Low Density I', 'Mixed Use',
       'Low Density II', 'High Density, Greenbelt', 'Institutional',
       'Low Density I, Greenbelt', 'Business Employment', 'Office',
       'Convenience Commercial', 'High Density, Mixed Use'], dtype=object)

In [14]:
df_res_centres[df_res_centres['OP_DES'] == 'Low Density I, Greenbelt']

FID  CENTRE_ID                  CHAR_AREA                    OP_DES  \
581  582       1355  Clarkson - Lorne Park NHD  Low Density I, Greenbelt   

    ZONING Z_AREA  N_OF_BLDGS  PARKING   Shape__Area  Shape__Length  \
581  R1-36      3           1       43  26778.964844    1004.267222   

                                              geometry  
581  POLYGON ((-79.62300 43.51242, -79.62297 43.512...

# Offices
## Office Buildings

In [ ]:
df_office_bldgs = gpd.read_file('https://opendata.arcgis.com/datasets/439debbd5bb142c98f834095e730f831_0.geojson')